### ISSUE TITLE
Odd jump in loss function

### ISSUE THREAD

**Issue by ImanHosseini** [pytorch/fairseq#3853]
I am using a transformer model, with **fixed** lr. 
```
fairseq-train --valid-subset test $DATA_LOC/data.toked.src-tgt --lr 0.01 --keep-last-epochs 1 --clip-norm 0.1 --dropout 0.5 --max-tokens 10000 --arch transformer --max-source-positions 2436 --max-target-positions 152 --encoder-attention-heads 16 --decoder-attention-heads 16 --save-dir $OUT_LOC/checkpoints --optimizer nag --tensorboard-logdir $OUT_LOC/tlogz
```
What I see in the loss function, is that after getting very close to zero (~0.05) there is a sudden jump.
![dp5_f](https://user-images.githubusercontent.com/12172889/132550408-988562d6-6dfa-4492-b849-2fb59ff32dd6.png)
What is going on here? Given that lr is fixed, why is that spike happening?
[latest version, but version seems to be irrelevant]
